dask_planner/src/sql/logical.rs

```diff
@@ -206,14 +208,14 @@ impl PyLogicalPlan {
 
     #[pyo3(name = "getRowType")]
     pub fn row_type(&self) -> PyResult<RelDataType> {
-        let schema = self.original_plan.schema();
-        let rel_fields: Vec<RelDataTypeField> = schema
-            .fields()
-            .iter()
-            .map(|f| RelDataTypeField::from(f, schema.as_ref()))
-            .collect::<Result<Vec<_>>>()
-            .map_err(py_type_err)?;
-        Ok(RelDataType::new(false, rel_fields))
+        let mut all_rel_fields: HashSet<RelDataTypeField> = HashSet::new();
```

Comment:
This is causing non-deterministic behavior. I assumed, incorrectly, that HashSet honored ordering but it does not. Will change to another datastructure.

Reply:
`itertools` has a `unique()` adaptor so maybe
```
use itertools::Itertools; # needs itertools as a cargo dependency
```
and
```rust
    pub fn row_type(&self) -> PyResult<RelDataType> {
        let all_rel_fields = self
            .original_plan
            .all_schemas()
            .into_iter()
            .flat_map(|schema| {
                schema
                    .fields()
                    .iter()
                    .map(|f| RelDataTypeField::from(f, schema.as_ref()))
            })
            .collect::<Result<Vec<_>>>()
            .map_err(py_type_err)?
            .into_iter()
            .unique()
            .collect::<_>();
        Ok(RelDataType::new(false, all_rel_fields))
    }
```

Reply:
Although it feels like there must be a way of doing that unwrapping of the `Iterator<Result<T, Err>>` without consing two new `Vec`s